auth/service.go

```diff
@@ -255,6 +255,15 @@ func (s *authService) GetAuthUserData(ctx context.Context, tenantToken TenantTok
 	return user.Data.Attributes, nil
 }
 
+func constructResolveTokenError(err error, res *http.Response, target string) error {
+	if res != nil && res.Header != nil {
+		if msg := res.Header["WWW-Authenticate"]; len(msg) > 0 {
```

Comment:
you could probably write something like: 
```
if msg, ok := res.Header["WWW-Authenticate"]; ok {
   ...
}
```

Reply:
also, can you write a "whitebox" test for this function, so you can test both cases (ie, an HTTP response from `auth` with and without the `WWW-Authenticate` header) ?

Reply:
if statement changed 131df61 

as for the whitebox test - this case is already covered by all tests that verify message of the format "error while resolving the token for" - eg this one: https://github.com/fabric8-services/fabric8-tenant/pull/689/files/588b0887f60d59fd269e4becc93f5621d507f112#diff-c18bfd11fd5d373e7c777c050f0bcf9eR44

Reply:
yes, I've seen this test, but you did not cover the case where the response does not contain the `WWW-Authenticate` header, which is why I suggested this "whitebox test". But if you add another test with a response without the aforementioned header, it'll be fine, too ;) As you like.

Reply:
ok, thanks! (sorry, I missed that in the existing code)